upgrade/upgradetest/config.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package upgradetest
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/upgrade"
+)
+
+// GetConfig returns an upgrade config with the provided fork scheduled to have
+// been initially activated and all other forks to be unscheduled.
+func GetConfig(fork Fork) upgrade.Config {
+	return GetConfigWithUpgradeTime(fork, upgrade.InitiallyActiveTime)
+}
+
+// GetConfig returns an upgrade config with the provided fork scheduled to be
+// activated at the provided upgradeTime and all other forks to be unscheduled.
+func GetConfigWithUpgradeTime(fork Fork, upgradeTime time.Time) upgrade.Config {
+	c := upgrade.Config{
+		ApricotPhase4MinPChainHeight: 0,
+		CortinaXChainStopVertexID:    ids.Empty,
+	}
+	// Initialize all forks to be unscheduled
+	SetConfigTimesTo(&c, Latest, upgrade.UnscheduledActivationTime)
+	// Schedule the requested forks at the provided upgrade time
+	SetConfigTimesTo(&c, fork, upgradeTime)
+	return c
+}
+
+// SetConfigTimesTo sets the upgrade time of the provided fork, and all prior
+// forks, to the provided upgradeTime.
+func SetConfigTimesTo(c *upgrade.Config, fork Fork, upgradeTime time.Time) {
```

Comment:
The signature looks like this to align with a pointer receiver method on the `Config` type (which can't be done here because this is a testing package)